program/src/processor.rs

```diff
@@ -2560,6 +2560,13 @@ impl Processor {
 
             if (side == Side::Bid && base_pos > 0) || (side == Side::Ask && base_pos < 0) {
                 0
+            } else if is_luna_market {
+                // Take into account outstanding open orders as well
+                let on_orders = match side {
+                    Side::Bid => mango_account.perp_accounts[market_index].bids_quantity,
```

Comment:
Ok, I see base_pos already took perpmarket.taker_base and filled maker orders into account.